core/parser.cpp

```diff
@@ -1186,9 +1186,42 @@ namespace {
 
                     Token op = pop();
                     if (op.kind == Token::BRACKET_L) {
-                        AST *index = parse(MAX_PRECEDENCE);
+                        const int max = 2;
+                        bool isSlice = true;
+                        AST *first = nullptr;
+                        AST *last = nullptr;
+
+                        for (int i = 0; i < max; i++) {
```

Comment:
I think this code would be simplified by unrolling the loop and folding the constant i values in.  You could handle the breaks by nesting ifs.